src/com/intellij/idea/plugin/hybris/properties/impl/PropertiesServiceImpl.kt

```diff
@@ -142,52 +104,114 @@ class PropertiesServiceImpl(val project: Project) : PropertiesService {
                     "env.properties" -> envPropsFile = file
                     "advanced.properties" -> advancedPropsFile = file
                     "local.properties" -> localPropsFile = file
-                    else -> {
-                        for (property in file.properties) {
-                            if (property.key != null) {
-                                result[property.key!!] = property
-                            }
-                        }
-                    }
+                    "project.properties" -> propertieFiles.add(file)
                 }
             }
-        addPropertyFile(result, envPropsFile)
-        addPropertyFile(result, advancedPropsFile)
-        addPropertyFile(result, localPropsFile)
 
-        val optDir = result[HybrisConstants.PROPERTY_OPTIONAL_CONFIG_DIR]
-        addOptionalConfiguration(result, optDir)
+        envPropsFile?.let { propertieFiles.add(0, it) }
+        advancedPropsFile?.let { propertieFiles.add(1, it) }
+        localPropsFile?.let { propertieFiles.add(it) }
+
+        propertieFiles.forEach { file -> addPropertyFile(result, file) }
+
+        loadHybrisRuntimeProperties(result)
+        loadHybrisOptionalConfigDir(result)
 
         return ArrayList(result.values)
     }
 
+    private fun findAllProperties(): LinkedHashMap<String, String> {
+
+        val result = LinkedHashMap<String, String>()
+        findAllIProperties().forEach { property ->
+            property.value?.let {
+                result[property.key!!] = it
+            }
+        }
+
+        replacePlaceholders(result)
 
+        return result
+    }
 
-    private fun addOptionalConfiguration(result: LinkedHashMap<String, IProperty>, optDir: IProperty?) {
-        if (optDir == null) {
-            return
+    private fun replacePlaceholders(result: LinkedHashMap<String, String>) {
+        result.forEach { key, value ->
+            if (value.contains(nestedPropertyPrefix)) {
+                replacePlaceholder(result, key, value);
+            }
         }
-        val dir = File(optDir.value)
-        if (!dir.isDirectory) {
-            return
+    }
+
+    private fun replacePlaceholder(result: LinkedHashMap<String, String>, key: String, value: String) {
+        val pattern = Pattern.compile("\\$\\{[^\\}]*\\}")
```

Comment:
`Pattern.compile` is quite heavy operation and it would be nice to extract it.

also, in Kotlin, it is possible to defined regex like that: `"..."..toRegex(RegexOption.IGNORE_CASE)`